packages/financial-templates-lib/src/price-feed/InsuredBridgePriceFeed.ts

```diff
@@ -0,0 +1,157 @@
+import { PriceFeedInterface } from "./PriceFeedInterface";
+import Web3 from "web3";
+import { BN } from "../types";
+import type { Logger } from "winston";
+import { InsuredBridgeL1Client, Relay } from "../clients/InsuredBridgeL1Client";
+import { InsuredBridgeL2Client, Deposit } from "../clients/InsuredBridgeL2Client";
+
+const { toBN, toWei } = Web3.utils;
+
+enum isRelayValid {
+  No, // Should be 0
+  Yes, // Should be 1
+}
+
+interface Params {
+  logger: Logger;
+  web3: Web3;
+  l1Client: InsuredBridgeL1Client;
+  l2Client: InsuredBridgeL2Client;
+}
+
+// Allows user to respond to a "relay" price request that was sent in response to a "deposit" on a InsuredBridge
+// deployed to an L2 network. The relay price request is submitted on L1. This pricefeed will respond True or False
+// based on whether the relay was correctly constructed to match a deposit. The price request includes parameters in
+// its ancillary data that must be parsed from the hex ancillary data.
+export class InsuredBridgePriceFeed extends PriceFeedInterface {
+  private readonly decimals: number;
+  private readonly logger: Logger;
+  private readonly web3: Web3;
+  private readonly l1Client: InsuredBridgeL1Client;
+  private readonly l2Client: InsuredBridgeL2Client;
+  private relays: Relay[] = [];
+  private deposits: Deposit[] = [];
+  private readonly convertPrice: (_number: number) => BN;
+
+  /**
+   * @notice Constructs the InsuredBridgePriceFeed.
+   * @param {Object} logger Winston module used to send logs.
+   * @param {Object} web3 Provider to connect to Ethereum network.
+   * @param {Object} l1Client Fetches and returns latest state of L1 pool and admin contracts.
+   * @param {Object} l2Client Fetches and returns latest state of L2 deposit contract.
+   */
+  constructor({ logger, web3, l1Client, l2Client }: Params) {
+    super();
+
+    this.decimals = 18;
+    this.logger = logger;
+    this.web3 = web3;
+    this.l1Client = l1Client;
+    this.l2Client = l2Client;
+    this.convertPrice = (_number) => toBN(toWei(_number.toString()));
```

Comment:
I think `convertPrice` is not a great name for this. `toBNWei` might be better. we do something like this here: https://github.com/UMAprotocol/protocol/blob/1fca52652a4995369bee2a89542f43396d502bd3/packages/trader/src/RangeTrader.ts#L6

Reply:
done